allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -223,6 +232,37 @@ def from_params(cls, vocab, params: Params) -> 'WikiTablesSemanticParser':
                    attention_function=attention_function)
 
 
+def _get_type_productions(vocab: Vocabulary, namespace: str) -> Dict[str, List[int]]:
+    """
+    Temporary method, until we get the type declaration (grammar) to build this for us.
+
+    This method takes all of the actions that we saw in the training data and constructs a
+    "grammar" from them, where here "grammar" means "valid productions for any type".  That is, if
+    we see the action "r -> [<e,r>, e]", we add "[<e,r>, e]" to the valid productions for type "r".
```

Comment:
You're actually adding the whole rule, not just the RHS.

Reply:
Done.